molecules_app/main.py

```diff
@@ -12,47 +11,28 @@ class Molecule(BaseModel):
     weight: str
 
 
-file_path = "C:\\Users\\gabi0\\Desktop\\chemical.txt"
-
-
 @app.post("/molecules")
-async def add_molecule(molecule: Molecule | list[Molecule]):
-    p = Path(file_path)
-    if not p.exists():
-        with p.open('w') as file:
-            content = json.dumps([])
-            file.write(content)
+async def add_molecule(molecule: Molecule | list[Molecule], file_path):
     with open(file_path, "r") as file:
         content = json.load(file)
-        content.append(molecule.model_dump())
+        if type(molecule) is list:
+            for mol in molecule:
+                content.append(mol.model_dump())
+        else:
+            content.append(molecule.model_dump())
     content = json.dumps(content)
     with open(file_path, "w") as file:
         file.write(content)
     return content
 
 
 @app.get("/molecules")
-async def fetch_molecules(name: str | None = None):
-    with open(file_path, "r") as file:
+async def fetch_molecules(file_path_add, name: str | None = None):
+    with open(file_path_add, "r") as file:
         content = json.load(file)
         if name:
             for mol in content:
-                if name == mol["name"]:
+                if mol["name"] == name:
                     return mol
         else:
             return content
```

Comment:
Tak jak wyżej, tylko tu sprawa jest nieco badziej skomplikowana. Trzeba zwrócić listę obiektów `Molecule`. W tym momencie zwracany obiekt ma typ 'list[dict]`, żeby przerobić go na `list[Molecule]`, możesz użyć tej samej składni `Molecule(**mol)`, ale w połączeniu z konstruktem w Pythonie nazywanym `list comprehension`, czyli metodą na tworzenie listy w jednej linijce. Leci to tak: `return [Molecule(**mol) for mol in content]`.

Funkcjonalnie jest to to samo co:

```
ret = []
for mol in content:
  ret.append(Molecule(**mol))
return ret
```